### PR TITLE
docs: Improve pointing docs

### PR DESCRIPTION
**Pull request by caksoylar**
A few miscellaneous tweaks to improve the newly added pointing-related docs:
- Add missing code fence languages
- Fix default `&msc` value in example
- Refactor hardware integration page so that tabs work fine
  - This made me rewrite the page so that the headers follow a semantic structure, rather than going file by file. This way we can have proper headers that apply for both unibody/split use cases.

Last one was discussed in the 2477 PR review. Feedback welcome on the approach, though I feel this is the best way. (Unfortunately the diff is hard to follow due to many moved chunks, so you might want to review the final page.)

## PR check-list

- [x] Branch has a [clean commit history](https://zmk.dev/docs/development/contributing/pull-requests#clean-commit-history)
- [ ] Additional tests are included, if changing behaviors/core code that is testable.
- [ ] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
- [x] [Pre-commit](https://zmk.dev/docs/development/local-toolchain/pre-commit) used to check formatting of files, commit messages, etc.

### CODE REVIEW

File: docs/docs/keymaps/behaviors/mouse-emulation.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ The mouse functionality will not work over BLE until that is done.
 
 To use any of the behaviors documented here, the ZMK mouse feature must be enabled explicitly via a config option:
 
-```
+```ini
```
[the remark this answers is not in the document]
Comment (caksoylar) [reply docs/docs/keymaps/behaviors/mouse-emulation.md:21]: I think we did in most places at least, yes. They were added in the initial PR by Joel to add syntax highlighting.

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 2]
```diff
@@ -1,30 +1,48 @@
...
+      { label: "Split Peripheral", value: "peripheral" },
+    ]}
+  >
+
+    {/* eslint-disable-next-line */}
+    {props.children}
+
+  </Tabs>
+
+);
```
[the remark this answers is not in the document]
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx:31]: As long as the selections are synced, I think having separate tabs isn't bad. I find it confusing when you select a tab and it affects things way down the page, and you can't switch back without scrolling to the top.

[the remark this answers is not in the document]
Comment (Nick-Munnich) [reply docs/docs/development/hardware-integration/pointing.mdx:31]: How about just moving the first tab with the trick and leaving the second untouched?

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 3]
```diff
@@ -43,55 +61,72 @@ For example, if setting up an [SPI device](https://github.com/zmkfirmware/zephyr
...
+To avoid duplicating work, this node can be defined in a common `.dtsi` file that is included into both central and peripheral `.overlay`/`.dts` files.
 
-Both peripheral and central make use of a `zmk,input-split` device, which functions differently depending on where it is used. To avoid duplicating work, this node can be defined in a common `.dtsi` file that is included into both central and peripheral `.overlay`/`.dts` files. Second, the input listener for the central side is added here, but disabled, so that keymaps (which are included for central and peripheral builds) can reference the listener to add input processors without issue.
+All split pointers are identified using a unique integer value, which is specified using the `reg` property and in the `@#` suffix for the node. If adding multiple peripheral pointers, be sure that each is given a unique identifier.
 
+Second, the input listener for the central side is added here, but disabled, so that keymaps (which are included for central and peripheral builds) can reference the listener to add input processors without issue.
```
[the remark this answers is not in the document]
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx]: I expanded it a bit, and added a note under node definition that describes how the user uses the node reference.

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 4]
```diff
@@ -1,30 +1,48 @@
 ---
```
[the remark this answers is not in the document]
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx:1]: I added a new section on configuration, seems useful to have it to me.

File: docs/docs/development/hardware-integration/pointing.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx]: Also added a new tab for split centrals vs. unibody, should be correct from build experiments I did, though I don't have hardware to test on.

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 5]
```diff
@@ -85,28 +75,82 @@ For example, if setting up an [SPI device](https://github.com/zmkfirmware/zephyr
...
         device = <&glidepoint>;
     };
 };
+```
+
+  </TabItem>
+  <TabItem value="central">
+
+### Shared Configuration
+
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:105]: I don't think I'm a fan of suggesting the use of a separate .dtsi file for the pointing stuff, it doesn't seem long enough to be worthwhile and putting it into the standard keyboard.dtsi avoids any "forgot to include" issues.

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 6]
```diff
@@ -85,28 +75,82 @@ For example, if setting up an [SPI device](https://github.com/zmkfirmware/zephyr
...
+The input listener that is used by the central side is added in a shared in a common `.dtsi` file that is included into both central and peripheral `.overlay`/`.dts` files, but disabled by default.
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:106]: "in a shared in a common"

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 7]
```diff
@@ -58,40 +98,80 @@ Every input device needs an associated listener added that listens for events fr
...
+    glidepoint_listener: glidepoint_listener {
         compatible = "zmk,input-listener";
-        device = <&glidepoint>;
-        input-processors = <&zip_xy_transform (INPUT_TRANSFORM_XY_SWAP | INPUT_TRANSFORM_X_INVERT | INPUT_TRANSFORM_Y_INVERT)>;
+        status = "disabled";
     };
 };
+```
+
+Users will use this node reference `&glidepoint_listener` in their keymaps if they want to add [input processors](#input-processors) to it.
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:118]: This sentence reads oddly

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 8]
```diff
@@ -58,40 +98,80 @@ Every input device needs an associated listener added that listens for events fr
...
+
+- Include the shared .dtsi file.
+- Enable the input listener that is created in our shared file.
+- Assign the pointing device node to the listener.
+
+```dts title="<central>.overlay"
+// Pull in the shared configuration
+#include "<keyboard>-pointing.dtsi"
+
+// Node added in Input Device section above
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:132]:
```suggestion
// Node from the previous Input Device section
```

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 9]
```diff
@@ -111,58 +191,135 @@ Input splits need to be nested under a parent node that properly sets `#address-
...
-- Add the device node for the physical pointer.
 - Update the input split with a reference to the new device node that should be proxied.
 
-```dts
-#include "common.dtsi"
+```dts title="<peripheral>.overlay"
+// Pull in the shared configuration
+#include "<keyboard>-pointing.dtsi"
 
+// Node added in Input Device section above
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:207]:
```suggestion
// Node from the previous Input Device section
```

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 10]
```diff
@@ -111,58 +191,135 @@ Input splits need to be nested under a parent node that properly sets `#address-
...
+  <TabItem value="unibody">
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_KEYBOARD
+
+# Other keyboard settings
+
+config ZMK_POINTING
+    default y
+
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:277]: I'd add some extra comments relating the SPI config flag to the pointing device here.

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 11]
```diff
@@ -111,58 +191,135 @@ Input splits need to be nested under a parent node that properly sets `#address-
...
+        device = <&glidepoint>;
+        input-processors = <&zip_xy_transform (INPUT_TRANSFORM_XY_SWAP | INPUT_TRANSFORM_X_INVERT | INPUT_TRANSFORM_Y_INVERT)>;
+    };
 };
+```
+
+## Configuration Setting
+
+If your keyboard hardware includes a pointing device by default, you can enable the [`ZMK_POINTING` config](../../config/pointing.md#general) in your keyboard definition.
+You can do that in your [`Kconfig.defconfig` file](new-shield.mdx#kconfigdefconfig), where you can also enable the config for the communication protocol used by the pointing device:
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:265]:
```suggestion
You can do that in your [`Kconfig.defconfig` file](new-shield.mdx#kconfigdefconfig), where you can also enable the config for the communication protocol (e.g. SPI, I2C) used by the pointing device:
```
Maybe also link the corresponding protocol to the corresponding Zephyr Kconfig flag docs?

File: docs/docs/development/hardware-integration/pointing.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nick-Munnich) [reply docs/docs/development/hardware-integration/pointing.mdx:32]: Happy with the tabs now as-is.

File: docs/docs/development/hardware-integration/pointing.mdx (hunk 10 above)
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx:277]: Can you elaborate what kind of comment you are thinking of?

File: docs/docs/development/hardware-integration/pointing.mdx (hunk 7 above)
Comment (caksoylar) [reply docs/docs/development/hardware-integration/pointing.mdx]: I reworded it, let me know if you can think of a clearer rewording.

File: docs/docs/development/hardware-integration/pointing.mdx (hunk 10 above)
Comment (Nick-Munnich) [reply docs/docs/development/hardware-integration/pointing.mdx:277]: Something like "If the above pointing device uses SPI", just to help avoid blind copypaste syndrome a bit more

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 12]
```diff
@@ -58,40 +98,80 @@ Every input device needs an associated listener added that listens for events fr
...
+  </TabItem>
+  <TabItem value="central">
 
-Some physical pointing devices may be generating input events that need adjustment before being sent to hosts. For example a trackpad might be integrated into a keyboard rotated 90° and need the X/Y data adjusted appropriately. This can be accomplished with [input processors](../../keymaps/input-processors/index.md). As an example, you could enhance the above listener with the following input processor that inverts and swaps the X/Y axes:
+### Shared Configuration
 
-```dts
-#include <dt-bindings/zmk/input_transform.h>
+The input listener that is used by the central side is added in the shared `.dtsi` file that is included into both central and peripheral `.overlay`/`.dts` files, but disabled by default.
+This is so that keymaps (which are included for both central and peripheral builds) can reference the listener to add input processors without failing with an undefined reference error.
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:107]:
```suggestion
The input listener that is used by the central side is added to the shared `.dtsi` file that is included into both central and peripheral `.overlay`/`.dts` files.

The input listener is disabled by default, and will be enabled later in the central part's overlay. This is so that keymaps (which are included for both central and peripheral builds) can reference the listener to add input processors without failing with an undefined reference error.
```

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 13]
```diff
@@ -58,40 +98,80 @@ Every input device needs an associated listener added that listens for events fr
...
+    glidepoint_listener: glidepoint_listener {
         compatible = "zmk,input-listener";
-        device = <&glidepoint>;
-        input-processors = <&zip_xy_transform (INPUT_TRANSFORM_XY_SWAP | INPUT_TRANSFORM_X_INVERT | INPUT_TRANSFORM_Y_INVERT)>;
+        status = "disabled";
     };
 };
+```
+
+If users want to add [input processors](#input-processors) to the listener node, they will use the node reference `&glidepoint_listener` in their keymaps.
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:118]:
```suggestion
If users want to add [input processors](#input-processors) to the listener node, they will use the corresponding node reference (here it is `&glidepoint_listener`) in their keymaps.
```

File: docs/docs/development/hardware-integration/pointing.mdx [hunk 14]
```diff
@@ -111,58 +191,135 @@ Input splits need to be nested under a parent node that properly sets `#address-
 };
 ```
 
-### Peripheral
+If users want to add [input processors](#input-processors) to the listener node, they will use the node reference `&glidepoint_listener` in their keymaps.
```
Comment (Nick-Munnich) [inline docs/docs/development/hardware-integration/pointing.mdx:194]:
```suggestion
If users want to add [input processors](#input-processors) to the listener node, they will use the corresponding node reference (here it is `&glidepoint_listener`) in their keymaps.
```

**Review by Nick-Munnich — APPROVED** [review]
LGTM!
